src/Agent.Plugins/PipelineCache/PipelineCacheServer.cs

```diff
@@ -136,9 +136,13 @@ public class PipelineCacheServer
                     
                     if (!string.IsNullOrEmpty(variableToSetOnHit))
                     {
-                        context.SetVariable($"{PipelineArtifactConstants.PipelineCache}.{variableToSetOnHit}", "True");
+                        if (context.Variables.ContainsKey(variableToSetOnHit))
+                        {
+                            context.Output($"Overwriting the 'variableToSetOnHit' value to {variableToSetOnHit}");
```

Comment:
variableToSetOnHit is not the name of the variable...

Reply:
and {variableToSetOnHit} is not the value.

{variableToSetOnHit} is the name
and
{{variableToSetOnHit}} (if that even works) is the previous value

Reply:
Yeah, changed the name to cacheHitVar, although {variableToSetOnHit} is the value 🤔

Reply:
Go over this with @owenhuynMSFT